site/documentation/content/api/device-connection/index.md

```diff
@@ -107,6 +107,176 @@ Implementors of this API may return a *404* status code in order to indicate tha
+
+| Name             | Mandatory | Location                 | AMQP Type    | Description |
+| :--------------- | :-------: | :----------------------- | :----------- | :---------- |
+| *content-type*   | no        | *properties*             | *string*     | MUST be set to `application/json` if the invocation of the operation was successful and the body of the response message contains payload as described below. |
+
+The result of a successful invocation is carried in a single Data section of the response message as a UTF-8 encoded string representation of a single JSON object. It is an error to include payload that is not of this type.
+
+The response message JSON object has the following properties:
+
+| Name                | Mandatory | JSON Type     | Description |
+| :------------------ | :-------: | :------------ | :---------- |
+| *adapter-instances* | *yes*     | *array*       | A non-empty array of JSON objects that represent the command-handling adapter instances along with the device or gateway id. |
+
+Each entry in the *adapter-instances* array has the following properties:
+
+| Name                  | Mandatory | JSON Type     | Description |
+| :-------------------- | :-------: | :------------ | :---------- |
+| *adapter-instance-id* | *yes*     | *string*      | The ID of the protocol adapter instance handling command & control messages for the device given in the *device-id* property. |
+| *device-id*           | *yes*     | *string*      | The ID of the gateway or device that the protocol adapter instance given by the *adapter-instance-id* is handling command & control messages for. |
+
+
+An example of a response message with a single adapter instance result, returned for example if an adapter instance is registered for the given device:
+
+~~~json
+{
+  "adapter-instances": [
+    {
+      "adapter-instance-id": "adapter-1",
+      "device-id": "4711"
+    }
+  ]
+}
+~~~
+
+An example of a response message with multiple contained adapter instances, returned for example if no adapter instance is registered for the given device or its last used gateway, and therefore a list of all adapter instances for the gateways of the device is returned:
+
+~~~json
+{
+  "adapter-instances": [
+    {
+      "adapter-instance-id": "adapter-1",
+      "device-id": "gw-1"
+    },
+    {
+      "adapter-instance-id": "adapter-1",
+      "device-id": "gw-2"
+    }
+  ]
+}
+~~~
+
+
+The response message's *status* property may contain the following codes:
+
+| Code | Description |
+| :--- | :---------- |
+| *200* | OK, the payload contains the gateway ID. |
+| *400* | Bad Request, the request message does not contain all required information/properties. |
+| *404* | Not Found, there is no command-handling adapter instance assigned to the device. |
+
+Implementors of this API may return a *404* status code in order to indicate that no device with the given identifier exists for the given tenant. However, performing such a check is optional.
```

Comment:
the table above seems to indicate that this check is actually mandatory for an implementation to perform ...

Reply:
The check mentioned in the table is a different one, actually.
The entry in the table makes it mandatory to return a 404 if not command-handling instance is found.
The extra hint below states that the implementation may first check if the given tenant actually has the given device. That would make for a more insightful error message if there was the wrong device id being used. However, this is probably a somewhat theoretical case since a client would usually invoke the "assert device registration" method shortly before and would get that error there already.

Reply:
I see, then what about starting like
> Implementors of this API may *also* return a *404* status ...